File: JAVA/src/main/java/com/inf_112/trabalho_final/service/MarcaService.java
package com.inf_112.trabalho_final.service;

import com.inf_112.trabalho_final.dao.MarcaDao;
import com.inf_112.trabalho_final.model.Marca;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MarcaService {

    private final MarcaDao MarcaDao;

    public MarcaService(MarcaDao MarcaDao) {
        this.MarcaDao = MarcaDao;
    }

    public Marca getById(int id) {
        return MarcaDao.findById(id);
    }

    public List<Marca> getAll() {
        return MarcaDao.findAll();
    }

    public void insert(Marca Marca) {
        MarcaDao.insert(Marca);
    }

    public void update(Marca Marca) {
        MarcaDao.update(Marca);
    }

    public void delete(int id) {
        MarcaDao.delete(id);
    }
}

File: C++/service/MarcaService.cpp
#include "MarcaService.h"

// Construtor
MarcaService::MarcaService(MarcaDao& dao) : marcaDao(dao) {}

// Obter Marca por ID
Marca MarcaService::getById(int id) {
    return marcaDao.findById(id);
}

// Obter todas as Marcas
std::vector<Marca> MarcaService::getAll() {
    return marcaDao.findAll();
}

// Inserir uma nova Marca
void MarcaService::insert(const Marca& marca) {
    marcaDao.insert(marca);
}

// Atualizar uma Marca existente
void MarcaService::update(const Marca& marca) {
    marcaDao.update(marca);
}

// Deletar uma Marca pelo ID
void MarcaService::deleteById(int id) {
    marcaDao.deleteById(id);
}
